docs/research/data_phases.md

```diff
@@ -0,0 +1,173 @@
+
+By delineating the lifecycle phases in which information is known, we can build a more modular library with tooling to optimize for code and data bundle sizes.
+
+## Three Phases of Information
+
+There are three points at which information may be known to the data provider:
+
+1. **Compile Time:** Based on the *programmer*. Can be determined by static analysis of code.
+2. **Construction Time:** Based on the *end user*. Can be determined from the locale, including Unicode extensions.
+3. **Format Time:** Based on the *runtime*. Can only be determined from the specific item being formatted (e.g., date or number).
+
+### Example: Duration Formatting
+
+Suppose you are formatting a time duration, such as a countdown timer for a video game. The breakdown of information known in each phase is:
+
+1. **Compile Time**
+    - Width (long, short, narrow, digital, …)
+2. **Construction Time**
+    - Locale
+    - Numbering system\*
+3. **Format Time**
+    - Specific set of time units (hour, minute, second, …)
+    - Plural form of the time units
+
+\* The numbering system determines the choice of digits. It does not currently affect the choice of unit pattern, but it may in the future.
+
+### Example: Date Formatting
+
+Date formatting is more complicated, but suppose you are formatting a date without a time based on a components bag. Information breakdown:
+
+1. **Compile Time**
+    - Coarse fields (months, weekdays, time zones, …)
+2. **Construction Time**
+    - Locale
+    - Calendar system\*
+    - Width\*\* (numeric, abbreviated, wide, …)
+3. **Format Time**
+    - Specific month, weekday, time zone, …
+
+\* The calendar system is special, because the program may or may not choose to support all calendar systems due to the amount of code and data that they require. Therefore, the calendar system may be known at compile time for certain applications.
+
+\*\* The width of fields can't be known until construction time because that might depend on the locale. For a given components bag, some locales may prefer wider month names whereas others may prefer narrower ones.
+
+## Resource Paths
+
+*Background reading: data_pipeline.md*
+
+The phase in which information is known should influence whether that information ends up in the resource key, resource options, or data struct. As a general rule of thumb:
+
+- If information is known at *compile time*, it should be in the *resource key*.
+- If information is known at *construction time*, it should be in the *resource options*.
+- If information is not known until *format time*, it should be in the data struct itself.
+
+Exceptions to the above rules of thumb could include:
+
+- If the amount of data is small, the pivot could be moved into the data struct. For example, different widths of currency symbols are often the same, so they could be combined into a single resource key, rather than having different resource keys for each width.
```

Comment:
"The pivot" is not defined anywhere in this document. Is it domain-specific jargon?

Reply:
Rephrased the sentence to remove the word "pivot"